test/EndToEnd/WorkloadTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+
+namespace EndToEnd.Tests
+{
+    public class WorkloadTests : TestBase
+    {
+        [Fact]
+        public void ItCannotPublishBlazorWasm_AOTWithoutWorkloadInstalled()
+        {
+            DirectoryInfo directory = TestAssets.CreateTestDirectory();
+            WorkloadTestEnvironment env = PrepareTestEnvironment(directory.FullName);
+
+            //FIXME: poke the value in the project file
+
+            // AOT builds are slow, so in case the test is failing (IOW, it is able to build)
```

Comment:
> AOT builds are slow, so in case the test is failing (IOW, it is able to build)

Then it should not be in the installer test. Installer end to end tests are for "if I have all the parts together" with the right version. dotnet/installer does not have parallel test support, and there are a lot of legs. It has less tolerant for flaky tests

Reply:
This is for testing whether all the parts for the workload, with the right versions work together or not. The tests shouldn't be flaky.

I'll change this to native relinking, which will be much quicker. And we can discuss AOT after that.